pkg/virt-api/webhooks/validating-webhook/admitters/pod-eviction-admitter.go

```diff
@@ -94,18 +94,22 @@ func (admitter *PodEvictionAdmitter) Admit(ctx context.Context, ar *admissionv1.
 
 	// This message format is expected from descheduler.
 	const evictionFmt = "Eviction triggered evacuation of VMI \"%s/%s\""
-	if markForEviction && !vmi.IsMarkedForEviction() && vmi.Status.NodeName == pod.Spec.NodeName {
-		err := admitter.markVMI(ctx, vmi.Namespace, vmi.Name, vmi.Status.NodeName, isDryRun(ar))
-		if err != nil {
-			// As with the previous case, it is up to the user to issue a retry.
-			return denied(fmt.Sprintf("kubevirt failed marking the vmi for eviction: %s", err.Error()))
+	if markForEviction {
+		if !vmi.IsMarkedForEviction() {
+			if vmi.Status.NodeName != pod.Spec.NodeName {
+				return denied("Eviction request for target Pod")
+			}
+			err := admitter.markVMI(ctx, vmi.Namespace, vmi.Name, pod.Spec.NodeName, isDryRun(ar))
+			if err != nil {
+				// As with the previous case, it is up to the user to issue a retry.
+				return denied(fmt.Sprintf("kubevirt failed marking the vmi for eviction: %s", err.Error()))
+			}
+			return denied(fmt.Sprintf(evictionFmt, vmi.Namespace, vmi.Name))
```

Comment:
I don't know if with early exit this is more readable:
```suggestion
	if !markForEviction {
		return validating_webhooks.NewPassingAdmissionResponse()
	}
	
	if vmi.IsMarkedForEviction() {
		return denied(fmt.Sprintf("Evacuation in progress: "+evictionFmt, vmi.Namespace, vmi.Name))
	}
	
	if vmi.Status.NodeName != pod.Spec.NodeName {
		return denied("Eviction request for target Pod")
	}
	
	err = admitter.markVMI(ctx, vmi.Namespace, vmi.Name, pod.Spec.NodeName, isDryRun(ar))
	if err != nil {
		// As with the previous case, it is up to the user to issue a retry.
		return denied(fmt.Sprintf("kubevirt failed marking the vmi for eviction: %s", err.Error()))
	}
	
	return denied(fmt.Sprintf(evictionFmt, vmi.Namespace, vmi.Name))
```
Up to you!

Reply:
Done.